### PR TITLE
fix issue #4779 - grey vs gray variables and classes

### PR DESCRIPTION
**Pull request by mcclowes**
### Issue Titles
[Variables] Added @gray and associated variants
[Various] class `.gray` works just like `grey`

### Closed Issues
#4779 

### Description
Grey is the English spelling, but is not consistent with CSS. Therefore, to avoid confusion, gray should be default, whilst still supporting grey for existing users.

### CODE REVIEW

**Comment by hammy2899** [comment]
This would create a breaking change, the best option here is to have `grey` and `gray` as options.

**Comment by mcclowes** [comment]
I made them both options

**Comment by mcclowes** [comment]
I did actually do that, it appears I just haven't built the css. Will do now.

**Comment by mcclowes** [comment]
@hammy2899 Built. This should not be a breaking change now

**Comment by hammy2899** [comment]
You have a merge conflict.

**Comment by mcclowes** [comment]
Yes I see that. I didn't touch `RELEASE-NOTES.md`, nor do I know how it's supposed to be structured so perhaps you could advise on that.

Are you satisfied that that the change is no longer a breaking change though?

**Comment by hammy2899** [comment]
Yes, the change is fine now.

Replace the release notes file with the one from the `next` branch that should solve the conflicts also try and remove the dist files from the commit.

**Comment by mcclowes** [comment]
Thanks, got rid of a load of unintended diffs.

File: src/themes/default/elements/icon.overrides [hunk 1]
```diff
@@ -36,11 +36,9 @@ i.icon.linkedin.in:before { content: "\f0e1"; }
 i.icon.zoom.in:before { content: "\f00e"; }
```
Comment (hammy2899) [inline src/themes/default/elements/icon.overrides]: Why was this file edited?

File: src/definitions/modules/search.js [hunk 2]
```diff
@@ -337,7 +337,7 @@ $.fn.search = function(parameters) {
               apiSettings = {
                 debug             : settings.debug,
                 on                : false,
-                cache             : true,
```
Comment (hammy2899) [inline src/definitions/modules/search.js]: Why was this changed?

File: package-lock.json [hunk 3]
```diff
@@ -1,102 +1,104 @@
 {
```
Comment (hammy2899) [inline package-lock.json]: This file shouldn't be added

File: src/definitions/modules/search.js (hunk 2 above)
Comment (mcclowes) [reply src/definitions/modules/search.js]: I think the issue is I branched off master rather than next

File: package-lock.json (hunk 3 above)
Comment (mcclowes) [reply package-lock.json]: sure

File: src/themes/default/elements/icon.overrides (hunk 1 above)
Comment (mcclowes) [reply src/themes/default/elements/icon.overrides]: I think the issue is I branched off master rather than next
